test/to.cpp

```diff
@@ -0,0 +1,330 @@
+#if 0 // P1141 syntax not supported
+		Same<std::vector<int>> auto a = ranges::to<std::vector<int>>(l);
+#else
+		auto a = ranges::to<std::vector<int>>(l);
+		static_assert(Same<decltype(a), std::vector<int>>);
+#endif
+		// Specify an allocator
+#if 0 // P1141 syntax not supported
+		Same<std::vector<int, Alloc>> auto b = ranges::to<std::vector<int, Alloc>(l, alloc);
+#else
+		auto b = ranges::to<std::vector<int, Alloc>>(l, alloc);
+		static_assert(Same<decltype(b), std::vector<int, Alloc>>);
+#endif
+		// copy a list to a vector of the same type, deducing value_type
+#if 0 // P1141 syntax not supported
+		Same<std::vector<int>> auto c = ranges::to<std::vector>(l);
+#else
+		auto c = ranges::to<std::vector>(l);
+		static_assert(Same<decltype(c), std::vector<int>>);
+#endif
+		// copy to a container of types ConvertibleTo
+#if 0 // P1141 syntax not supported
+		Same<std::vector<long>> auto d = ranges::to<std::vector<long>>(l);
+#else
+		auto d = ranges::to<std::vector<long>>(l);
+		static_assert(Same<decltype(d), std::vector<long>>);
+#endif
+		// Supports converting associative container to sequence containers
+#if 0 // P1141 syntax not supported
+		Same<std::vector<std::pair<const int, int>>>
+			auto f = ranges::to<vector<std::pair<const int, int>>>(m);
+#else
+		auto f = ranges::to<std::vector<std::pair<const int, int>>>(m);
+		static_assert(Same<decltype(f), std::vector<std::pair<const int, int>>>);
+#endif
+		// Removing the const from the key by default
+#if 0 // P1141 syntax not supported
+		Same<std::vector<std::pair<int, int>>> auto e = ranges::to<std::vector>(m);
+#else
+		auto e = ranges::to<std::vector>(m);
+#if 0 // Per the Kona minutes: "[stripping of const from first element in pair should be removed]"
+		static_assert(Same<decltype(e), std::vector<std::pair<int, int>>>);
+#else
+		static_assert(Same<decltype(e), std::vector<std::pair<const int, int>>>);
+#endif
+#endif
+		// Supports converting sequence containers to associative ones
+#if 0 // P1141 syntax not supported
+		Same<std::map<int, int>> auto g = f | ranges::to<map>();
+#else
+		auto g1 = f | ranges::to<std::map>();
+		static_assert(Same<decltype(g1), std::map<int, int>>);
+#endif
+		// Pipe syntax
+#if 0 // P1141 syntax not supported
+		Same<std::vector<int>> auto g = l | ranges::view::take(42) | ranges::to<std::vector>();
+#else
+		auto g2 = l | ranges::view::take(42) | ranges::to<std::vector>();
+		static_assert(Same<decltype(g2), std::vector<int>>);
+#endif
+		// Pipe syntax with allocator
+		auto h = l | ranges::view::take(42) | ranges::to<std::vector>(alloc);
+#if 1 // Additional validation
+		static_assert(Same<decltype(h), std::vector<int, Alloc>>);
+		CHECK(h.get_allocator().x == 42);
+#endif
+		// The pipe syntax also support specifying the type and conversions
+		auto i = l | ranges::view::take(42) | ranges::to<std::vector<long>>();
+		// Parentheses are optional for template...
+#if 0 // P1141 syntax not supported

```

Comment:
Why?

Reply:
I suspect that my comment is unclear, and that you're not actually asking me to explain why GCC doesn't yet support P1141's syntax for placeholder types. I'll clarify.